api/src/test/java/org/apache/iceberg/expressions/TestInclusiveManifestEvaluator.java

```diff
@@ -108,6 +112,9 @@ public void testAllNulls() {
 
     shouldRead = ManifestEvaluator.forRowFilter(startsWith("all_nulls", "asad"), SPEC, true).eval(FILE);
     Assert.assertFalse("Should skip: startsWith on all null column", shouldRead);
+
+    shouldRead = ManifestEvaluator.forRowFilter(notStartsWith("all_nulls", "asad"), SPEC, true).eval(FILE);
+    Assert.assertFalse("Should skip: notStartsWith on all null column", shouldRead);
```

Comment:
In line with the definition of `notStartsWith` against `null` had with @rdblue in the `testStringNotStartsWith` method being added to this file, we've decided to define `notStartsWith` aginast `null` to be `true`. In this case, `true` means that the file is maybe a match. Because Iceberg does not implement SQL's 3-way boolean logic, we have to make a choice here. It makes rational sense to say that `null` does not start with some non-null string, and returning `true` here allows for the query engine to then further process the file and make its own decisions according to its own customs (such as whether the query engine uses SQL 3-way boolean logic or not). If we return less data, the query engine does not have the opportunity to return all possible results.